lib/indexer/filterindexer.js

```diff
@@ -59,12 +62,12 @@ class FilterIndexer extends Indexer {
   async indexBlock(meta, block, view) {
     const hash = block.hash();
     const prev = await this.getFilterHeader(block.prevBlock);
-    const basic = block.toFilter(view);
+    const basic = block.toFilter(view, this.filterType);
 
     const filter = new Filter();
     filter.header = basic.header(prev);
     filter.filter = basic.toRaw();
-    await this.blocks.writeFilter(hash, filter.toRaw());
+    await this.blocks.writeFilter(hash, filter.toRaw(), this.filterType);
```

Comment:
I think we should rename the variable `basic` here to something else, maybe just `filter` because it may not be BASIC all the time!